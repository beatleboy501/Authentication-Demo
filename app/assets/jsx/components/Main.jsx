import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import Form from './Form.jsx';
import Sentiment from './Sentiment.jsx';
import axios from 'axios';

export default class Main extends Component {
  constructor(props) {
    super(props);
    this.validate = this.validate.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
    this.handlePasswordChange = this.handlePasswordChange.bind(this);
    this.apiRoot = "https://beatleboy501-authenticate-demo.herokuapp.com/api"; //"http://localhost:8080/api";
    this.state = {
      username: "",
      password: ""
    }
  }

  validate(data) {
    let isValid = true;
    let invalidEntries = [];
    Object.keys(data).forEach((key) => {
      let value = data[key];
      if (0 === value.length || !value) {
        isValid = false;
        invalidEntries.push(key.charAt(0).toUpperCase() + key.slice(1));
      }
    });
    return {
      isValid: isValid,
      invalidEntries: invalidEntries
    };
  }

  handleSubmit(e) {
    let self;
    e.preventDefault();
    self = this;
    const data = {
      username: this.state.username,
      password: this.state.password
    };
    const validationObject = this.validate(data);
    if (!validationObject.isValid) {
      alert(`Please Check the Following Entries:\n${ validationObject.invalidEntries }`);
      return;
    }
    let url = `${this.apiRoot}/create`;
    this.setState({
      username: "",
      password: ""
    });
    e.target.reset();
    axios.post(url, data)
        .then(data => {
          alert('User saved successfully');
        })
        .catch(err => {
          alert(err.data.message);
        });
  }

  handleInputChange(e) {
    this.setState({
      username: e.target.value
    })
  }

  handlePasswordChange(e) {
    this.setState({
      password: e.target.value
    })
  }

  render() {
    return (
        <div className="container">
          <ul className="nav nav-tabs">
            <li className="active"><a data-toggle="tab" href="#menu1">Home</a></li>
            <li><a data-toggle="tab" href="#menu2">Create User</a></li>
            <li><a data-toggle="tab" href="#menu3">Sentiment</a></li>
          </ul>
          <div className="tab-content">
            <div id="menu1" className="tab-pane fade in active">
              <h3>Home</h3>
              <p>The Create User tab is used to create a new User.</p>
              <p>The Sentiment tab is used to retrieve a "sentiment" of good, bad, or neutral as well as a confidence
                score.</p>
              <p>Server documentation can be found <a href="https://beatleboy501-authenticate-demo.herokuapp.com/module-Server.html">here</a>. </p>
            </div>
            <div id="menu2" className="tab-pane fade">
              <h3>Create User</h3>
              <Form identifier="createForm" onSubmit={this.handleSubmit}
                    onInputChange={this.handleInputChange} onPasswordChange={this.handlePasswordChange}
                    username={this.state.username} password={this.state.password}></Form>
            </div>
            <div id="menu3" className="tab-pane fade">
              <h3>Sentiment</h3>
              <Sentiment apiRoot={this.apiRoot} validation={this.validate}></Sentiment>
            </div>
          </div>
        </div>
    );
  }
}
