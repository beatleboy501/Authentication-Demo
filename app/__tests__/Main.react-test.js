import React from 'react';
import Main from '../assets/jsx/components/Main.jsx';
import renderer from 'react-test-renderer';

test('Main renders', () => {
  const component = renderer.create(<Main></Main>);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
