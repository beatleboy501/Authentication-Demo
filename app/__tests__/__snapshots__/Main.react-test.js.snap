// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders 1`] = `
<div
  className="container"
>
  <ul
    className="nav nav-tabs"
  >
    <li
      className="active"
    >
      <a
        data-toggle="tab"
        href="#menu1"
      >
        Home
      </a>
    </li>
    <li>
      <a
        data-toggle="tab"
        href="#menu2"
      >
        Create User
      </a>
    </li>
    <li>
      <a
        data-toggle="tab"
        href="#menu3"
      >
        Authenticate User
      </a>
    </li>
    <li>
      <a
        data-toggle="tab"
        href="#menu4"
      >
        Sentiment
      </a>
    </li>
  </ul>
  <div
    className="tab-content"
  >
    <div
      className="tab-pane fade in active"
      id="menu1"
    >
      <h3>
        Home
      </h3>
      <p>
        The Create User tab is used to create a new User.
      </p>
      <p>
        The Authenticate User tab is used to generate the Auth token.
      </p>
      <p>
        The Sentiment tab is used to retrieve a "sentiment" of good, bad, or neutral as well as a confidence score.
      </p>
      <p>
        Server documentation can be found 
        <a
          href="https://authenticate-demo.herokuapp.com/module-Server.html"
        >
          here
        </a>
        . 
      </p>
    </div>
    <div
      className="tab-pane fade"
      id="menu2"
    >
      <h3>
        Create User
      </h3>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-3"
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <label>
                Username:
              </label>
              <br />
              <input
                onChange={[Function]}
                type="text"
                value={undefined}
              />
              <br />
            </div>
            <div>
              <label>
                Password:
              </label>
              <br />
              <input
                onChange={[Function]}
                type="password"
                value={undefined}
              />
              <br />
            </div>
            <br />
            <button
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          className="col-md-3"
        />
      </div>
    </div>
    <div
      className="tab-pane fade"
      id="menu3"
    >
      <h3>
        Authenticate User
      </h3>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-3"
        >
          <form
            onSubmit={[Function]}
          >
            <div>
              <label>
                Username:
              </label>
              <br />
              <input
                onChange={[Function]}
                type="text"
                value={undefined}
              />
              <br />
            </div>
            <div>
              <label>
                Password:
              </label>
              <br />
              <input
                onChange={[Function]}
                type="password"
                value={undefined}
              />
              <br />
            </div>
            <br />
            <button
              type="submit"
            >
              Submit
            </button>
          </form>
        </div>
        <div
          className="col-md-3"
        />
      </div>
    </div>
    <div
      className="tab-pane fade"
      id="menu4"
    >
      <h3>
        Sentiment
      </h3>
      <div
        className="row"
      >
        <div
          className="col-md-3"
        />
        <div
          className="col-md-3"
        >
          <form
            onSubmit={[Function]}
          >
            <label>
              Auth Token:
            </label>
            <br />
            <textarea
              cols="50"
              name="token"
              onChange={[Function]}
              rows="4"
              value=""
            />
            <br />
            <label>
              Sentence:
            </label>
            <br />
            <textarea
              cols="50"
              name="sentence"
              onChange={[Function]}
              rows="4"
              value=""
            />
            <br />
            <button
              type="submit"
            >
              Get Sentiment
            </button>
          </form>
        </div>
        <div
          className="col-md-3"
        />
      </div>
    </div>
  </div>
</div>
`;
